src/main/java/com/fasterxml/jackson/databind/deser/BasicDeserializerFactory.java

```diff
@@ -827,9 +827,16 @@ public JsonDeserializer<?> createCollectionDeserializer(DeserializationContext c
                 if (implType == null) {
                     // [databind#292]: Actually, may be fine, but only if polymorphich deser enabled
                     if (type.getTypeHandler() == null) {
-                        throw new IllegalArgumentException("Cannot find a deserializer for non-concrete Collection type "+type);
+                        // [databind#4783] Since 2.19, Allow `@JsonMerge` with Custom Collection extension.
+                        // Let's not throw exceptino so instantiation can happen downstream
+                        if (_canMapSuperInterfaceAsAbstractCollectionType(type, config)) {
```

Comment:
.

Reply:
I tried actually, it fails with below exception.
I checked what gets created inside `AbstractDeserializer` and almost nothing to actually deserialize anything.
The intent behind the `// do nothing` current approach was to create  `StringCollectionDeserializer` / `CollectionDeserializer` below starting at line 862.

```text
com.fasterxml.jackson.databind.exc.InvalidDefinitionException: Cannot construct instance of `com.fasterxml.jackson.databind.tofix.JsonMergeWithCustomCollection4783Test$MyListCustom` (no Creators, like default constructor, exist): abstract types either need to be mapped to concrete types, have custom deserializer, or contain additional type information
 at [Source: REDACTED (`StreamReadFeature.INCLUDE_SOURCE_IN_LOCATION` disabled); line: 1, column: 11] (through reference chain: com.fasterxml.jackson.databind.tofix.JsonMergeWithCustomCollection4783Test$MergeCustomStrin
	```

Reply:
But wouldn't hurt showcasing here! Lemme just make the suggested change so we both can see

Reply:
Yup, you can see checks are failing with the same exceptions.